src/requestApproval.ts

```diff
@@ -0,0 +1,58 @@
+import { S3 } from 'aws-sdk'
+import * as fs from 'fs';
+import * as cli from 'cli-color';
+import { Md5 } from 'ts-md5/dist/md5'
+
+import {Arguments} from 'yargs';
+import {loadStackArgs, createStackMain} from './cfn/index';
+import configureAWS from './configureAWS';
+
+export async function requestApproval(argv: Arguments): Promise<number> {
+  await configureAWS('sandbox', 'us-east-1');
+
+  const stackArgs = await loadStackArgs(argv);
+  const s3        = new S3();
+  const md5       = new Md5();
+
+  var approvedBucket = bucketLocation( stackArgs.ApprovedTemplateLocation )
+  var pendingBucket  = bucketLocation( stackArgs.PendingTemplateLocation )
+
+  var cfnTemplate = await fs.readFileSync(stackArgs.Template)
+  var key         = stackArgs.Template.replace("./","")
+  var hashedKey   = md5.appendStr(key).end()
+
+  var putObjectParams  = <Params> {Body: cfnTemplate, Bucket: pendingBucket, Key: `${stackArgs.StackName}/${hashedKey}.yaml`}
+  var headObjectParams = <Params> {Bucket: approvedBucket, Key: `${stackArgs.StackName}/${hashedKey}.yaml`}
+
+  try {
+    await s3.headObject(headObjectParams, undefined).promise()
+
+    logSuccess(`👍 Your template has already been approved`)
+  } catch(e) {
+    if ( e["code"] == "NotFound" ) {
+      await s3.putObject(putObjectParams, undefined).promise()
+
+      logSuccess(`Successfully uploaded ${key} to ${pendingBucket}`)
+      logSuccess(`Approve template with \`iidy approve ${hashedKey}.yaml\``)
+    } else {
+      console.log(e)
+      return 1
+    }
+  }
+
+  return 0
+}
+
+function bucketLocation(bucket : string): string {
+  if (bucket === undefined || bucket.length <= 0 ) {
+    throw new Error("Missing key: `ApprovedTemplateLocation` and `PendingTemplateLocation` cannot be empty")
+  }
+  // buckets have 's3://' prepend
+  return bucket.substr(5).slice(0,-1)
```

Comment:
The code above assumes it will only include a bucket name (and assumes it will start with `s3://` and end with `/` without checking for either). I think `ApprovedTemplateLocation` and `PendingTemplateLocation` should support having a path in addition to the bucket name to aid in organizing the template files on S3. If not, then drop the protocol and have it just be `ApprovedTemplateBucket`.